src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -21,29 +21,19 @@ class OrderedRVD private(
   self =>
   def rowType: TStruct = typ.rowType
 
-  def insert[PC](newContext: () => PC)(typeToInsert: Type,
-    path: List[String],
-    // rv argument to add is the entire row
-    add: (PC, RegionValue, RegionValueBuilder) => Unit): OrderedRVD = {
-    val localTyp = typ
-
-    val (insTyp, inserter) = typ.insert(typeToInsert, path)
-    OrderedRVD(insTyp,
-      partitioner,
-      rdd.mapPartitions { it =>
-        val c = newContext()
-        val rv2b = new RegionValueBuilder()
-        val rv2 = RegionValue()
+  // should be totally generic, permitting any number of keys, but that requires more work
+  def downcastToPK(): OrderedRVD = {
+    val newType = new OrderedRVType(partitionKey = typ.partitionKey,
+      key = typ.partitionKey,
+      rowType = rowType)
+    OrderedRVD(newType, partitioner, rdd)
+  }
 
-        it.map { rv =>
-          val ur = new UnsafeRow(localTyp.rowType, rv)
-          rv2b.set(rv.region)
-          rv2b.start(insTyp.rowType)
-          inserter(rv.region, rv.offset, rv2b, () => add(c, rv, rv2b))
-          rv2.set(rv.region, rv2b.end())
-          rv2
-        }
-      })
+  def upcast(castKeys: Array[String]): OrderedRVD = {
```

Comment:
And upcast is generally a checked operation ... maybe these are OK.

Reply:
doesn't this constructor check ordering?

Reply:
yeah, it totally does.

Reply:
Yes, it does.  That's what I meant when maybe these names were OK (analogies).